### ISSUE TITLE
Support Chuangmi 720p camera?

### ISSUE THREAD

**Issue by jymbob** [samtap/fang-hacks#235]
Just picked up a pair of "Chuangmi" cameras from GearBest - https://www.gearbest.com/ip-cameras/pp_701750.html

They're using the same app, and appear to be very similar in setup.

My FTDI is in the post so I may be able to provide some debugging information if that's helpful.

**Comment by VINCE43** [comment]
Hi, 
I just try tonight, and it doesn't work .. :-(

**Comment by bernardo-amaral** [comment]
I wanna hack this model too

**Comment by chris777c** [comment]
I have this model too, I can help you if you need anything tell me

**Comment by samtap** [comment]
- Post teardown pictures
- Intercept firmware update
- Attach serial console to capture boot logs, dump filesystem, crack root passwd etc.
- Find SDK, code examples etc.

**Comment by chris777c** [comment]
OK I undesstand I am so enthusiastic, in my level :

- Post teardown pictures ==> OK, I can do that

- Intercept firmware update ==> How to do that ? with what tool ?

**Comment by jymbob** [comment]
I'll see if I can get anything out of the serial console this evening

**Comment by chris777c** [comment]
Ok thank you

**Comment by jymbob** [comment]
Hm. Think I've identified the RX and TX pins but my soldering's appalling, so I haven't got to a console yet.
It looks like it's using the same SoC as the Mijia 360 720p, which [this project](https://github.com/Filipowicz251/mijia-1080P-hacks) has working, however I couldn't get ssh access working on mine, and reflashing the 360 firmware appears to have left me with a non-functional camera - I still have a working one, so I'll persevere for a bit longer!

**Comment by chris777c** [comment]
Thank you for you work

**Comment by rap1** [comment]
Hi,
I also have  Chuangmi 720p camera  and would like to make it work locally, for example through RTSP, without Internet connection. 
Do you have any update?
thank you

**Comment by ras31415** [comment]
Hi;
I also have 3 pcs :-(
I was little bit stupid and I didn't think that it is a closed system without open streaming protocolls.
Thanks before

**Comment by rap1** [comment]
the same for me: I found out that it works only through Internet.
I hope that, as already done for other Xiaomi cameras, it's possible to load some kind of RTSP server to stream the video locally. I look forward to receiving some good news...

**Comment by LarryN3t** [comment]
Serial interface connected
MAIN CHIP GM8135S
Wireless Chip realtek 8189

need support to develop img

**Comment by rap1** [comment]
Hi Larry, any news?
How did you succeed in connecting serial interface ? 
thank you

**Comment by LarryN3t** [comment]
Yes... connected and hacked.

**Comment by ras31415** [comment]
This is very joyful.
Could you share the solution of the hacking with we?

**Comment by rap1** [comment]
Great !  Could you give us details ?
Is it possible to load and run a local RTSP server ?

**Comment by sibojia** [comment]
@LarryN3t Could you share the solution? I also wish to connect to this camera. Thanks a lot!

**Comment by scapman** [comment]
Very great news!
Could you please share the solution?
It looks like you are the only one that made it.

Thanks a lot!

**Comment by lilipfem** [comment]
Please, please share your solution if one exist  :-/

This camera make me crazy ...

**Comment by bernardo-amaral** [comment]
anyone had solved this?

**Comment by ril3y** [comment]
I ordered one.  I will get FH working on if its possible.  However gearbest
takes a few weeks to get it.

**Comment by LarryN3t** [comment]
Hi all,
sorry for delay, this is my hack procedure for chuangmi 720p:

Connect the camera by phone app, Mi home, and configure wireless connetion to your wireless network.
Your camera will be reached by cloud.

First Hack:
Look the IP address assigned to camera by your wireless router, try to telnet: 
Username root 
Password blank

enjoy  

For the recovery image i will be share you the next week, i need to check if are present my personal configurations insight.

i want to suggest you the path to complete the hack:

After login by telnet, you can copy memory camera partitions on SD Card:

# fdisk -l 

First partition is "bootloader"
Second partition is "Kernel"
Third partition is  "Filesystem"
fourth partition is "Data" 
fifth partition "Checksum"

you can create recovery image with:

cat Kernel Filesystem Data Checksum > tf_recovery.img

If you try to create the hack alone, use other xiaomi camera hack and compare your Filesystem with another hack.

"Filesystem" image can be mounted on yor PC.

See you later.

Larry

**Comment by chris777c** [comment]
Thank you for feedback but I tried telnet with putty, his show me "Connection refused"

**Comment by ras31415** [comment]
Yes, absolutely same happened with me too.
telnet - l root 192.168.3.101 23
telnet: unable to connect to remote host: Connection refused.

**Comment by LarryN3t** [comment]
this is original image, put in sd and turn on the camera
https://drive.google.com/file/d/1xuBw8I_-rHEyBcXRPGjyBLPwiEpusU6B/edit
With this i connect by telnet

**Comment by chris777c** [comment]
Thank you, it's work, I am extract img from camera.

**Comment by lilipfem** [comment]
Where do you take the .img ? The link don't work any more  :-(

**Comment by ras31415** [comment]
Where did LarryN3t disappear?
His img is working.

**Comment by lilipfem** [comment]
So please, where can I found it ? I'm a poor newbies completly lost

**Comment by ras31415** [comment]
This is a link where you can download larryN3t's image for 24 hours. Unfortunetelly my dropbox is full so I can't keep it there.

https://www.dropbox.com/s/qm4squv21ik4bob/tf_recovery.img?dl=0

**Comment by lilipfem** [comment]
Thank you very much

**Comment by jymbob** [comment]
Did a quick fork and stuck larryN3t's file [here](https://github.com/jymbob/fang-hacks) - untested.

**Comment by rap1** [comment]
Thank you , I 'm reading on your link the procedure to load the new image.  
Has anybody tested it ?

**Comment by rap1** [comment]
@jymbob when you say "For Chuangmi 720 camera download tf_recovery.img, transfer to an SD card and boot" ,  do you mean to copy " tf_recovery.img" file on the SD without uncompressing  it  , correct? 
Should we prepare an sd-card with two partitions as explained below  or is not needed ?
Could you please clarify?
Thank you very much

**Comment by jymbob** [comment]
I haven't actually tried it yet, just repeating what the OP wrote here. I
will endeavour to try it and update the readme. As I said, my main concern
was making sure the image was accessible somewhere.

If anyone has any feedback, I'm happy to update the readme.

**Comment by lilipfem** [comment]
Try 
https://sourceforge.net/projects/win32diskimag to er/files/latest/download
To copy the .img to the SD card
Choose the file, the destination, and click write
I'm trying to see how tell the cam to boot to the SDcard

**Comment by rap1** [comment]
Thank you lilipfem ! Let us know if you succeed in booting from the SD card

**Comment by lilipfem** [comment]
No it's does'nt work. It's not a boutable image  :-(

**Comment by jymbob** [comment]
Found some time.
You don't need to unpack the image. Simply put it on the root of an SD
card. The camera checks for this file on bootup

Steps to get telnet access:
1. Set up camera with MiHome app
2. Go into details and make a note of camera IP address (e.g. 192.168.1.250)
3. Copy tf_recovery.img to the root of an SD card
4. Unplug camera
5. Insert SD card
6. Plug in camera
7. wait for about 2 minutes until lights go blue
8. telnet 192.168.1.250 (same IP as in step 2) username: root, no password

Given we now have root access, we should hopefully be able to run a local
RTSP server.

**Comment by rap1** [comment]
Great news !  Thank you very much for sharing the procedure :)
Maybe the RTSP server is already present in the image  (url: rtsp://device-ip/unicast) , could it be ?

**Comment by lilipfem** [comment]
Thank you @jymbob for your help.
I can now open tenet on the cam and that's all  :-(
i answer root for the login and after only a # prompt
Everywhere i saw that i would have to connect with a url like http://192.168.1.19/cgi-bin/status

Else i try with VideoStation of Synology and vlc and nothing work

**Comment by rap1** [comment]
..how did you format the SD ?  FAT16, FAT32 or other ?

**Comment by lilipfem** [comment]
If the question is for me, FAT32 ...

**Comment by rap1** [comment]
I also have a FAT32 formatted SD and now I can also open telnet on the cam :)
Now we should understand how to run a local RTSP server..

**Comment by joshdinsdale** [comment]
Following this with anticipation! I have gotten telnet working, whats involved in getting an rtsp server running, i know a bit of linux stuff but not so much about these prebuilt firmware images. Happy to help if anyone can point me in the right direction.

**Comment by tomsteenbakkers** [comment]
I’m also stuck after i got telnet access. 

1. I format a SD card with FAT32
2. Put the image tf_recovery.img on it. 
3. Switch off the camera. 
4. Put the SD card in the camera and power it on. 
5. When the blue light stopped blinking I checked with Fing and now the telnet protocol is there on port 23. 

Next 

1. I removed the card
2. downloaded the Image to write directly to the sd-card using Win32 Disk Imager. 
3. The I removed the power from the camera
4. Put the power back on 
5. when the blue light stopped blinking I inserted the card. 

But when I used http://192.168.1.9/cgi-bin/status nothing happens. Also http://192.168.1.9/cgi-bin/hello.cgi is not doing anything. 

However, I can make a telnet connection. Is there a way to activate the hack using telnet? Or I’m doing something wrong. The current firmware on my camera is 3.3.6_2017092810 and the camera is still working using the app

**Comment by jymbob** [comment]
The fang hack is designed for different firmware. I'm talking with the guys
who hacked the mijia 1080 which is similar firmware. It looks like we need
to recompile the rtsp library against the SoC SDK. I've had some success
getting a few things running manually on the camera (ssh, httpd) but we're
still a way off a nice turnkey solution. There are a lot of parts to the
process which are new to me, and I've only got limited time available. If
anyone has experience creating uboot images that'd be a great help.

**Comment by rap1** [comment]
I noticed that on Fang hacks, they use "snx_rtsp_server", we could try to use the same executable and see if it runs,  or if we need to recompile it , where can we find the SoC SDK ?
@jymbob did you have any news from the guys who hacked the mijia 1080 ?

**Comment by jymbob** [comment]
I've found what looks like an SDK for the Grain Media chipset (I don't know any Chinese). A search for "gm813x sdk" should give you some download links. The rtsp server for the mijia 1080p is [here](https://github.com/willthrom/rtspServer_gm)

**Comment by lilipfem** [comment]
Is there any news ? A solution founded ?

**Comment by rap1** [comment]
not yet, has anyone found a solution for loading RTSP server ?

**Comment by kuchy** [comment]
Hi any progress with RTSP server?

**Comment by scapman** [comment]
Hi. Any good news?

**Comment by alexzip87** [comment]
no news bad news :(

**Comment by jymbob** [comment]
No news no news. Sorry, no time to spend on this at the moment.

**Comment by sdesalve** [comment]
Hi @jymbob, is there any news? 

Have you founded time to run RTSP server on our Chuangmi Cam?

**Comment by zloybatanik** [comment]
I found! at first you must use recovery image https://github.com/jymbob/fang-hacks
and then install this https://github.com/ghoost82/mijia-720p-hack

rtsp working with 3-4 sec lag

**Comment by sdesalve** [comment]
@zloybatanik 
not worked for me
My Chunagmi firmware is 3.3.6_2017092810

Mi Home suggest me an UPDATE: 3.3.6_201802014. Should I apply them?

**Comment by zloybatanik** [comment]
@sdesalve  try update 3.3.6_201802014 first

**Comment by sdesalve** [comment]
@zloybatanik doesn't work for me.

I've used recovery image (https://github.com/jymbob/fang-hacks) that downgrade firmware to 3.3.6_2017092810 and let me login thru telnet.

Then I've installed this https://github.com/ghoost82/mijia-720p-hack but without lucky: is there only a telnet server and mijia-720p-hack.cfg is ignored...

I've also tried to upgrade firmware to 3.3.6_2017092810 with MiHome but same result: don't work 
:(

**Comment by zloybatanik** [comment]
Did same with my second camera, works perfectly.
My steps:
1. I format a SD card with FAT32
2. Put the image tf_recovery.img on it.
3. Switch off the camera.
4. Put the SD card in the camera and power it on.
5. Switch off the camera.
6. Copy everything from https://github.com/ghoost82/mijia-720p-hack/tree/master/sdcard to SD card root
7. Edited mijia-720p-hack.cfg
8. Put the SD card in the camera and power it on.
9. When the blue light stopped blinking open http://camera_ip/ (don`t try /cgi-bin/status, its working from the "/")

**Comment by tomsteenbakkers** [comment]
Just checking. This is the camera you used for the hack?

[image1.jp2]

Tom

On 25 Aug 2018, at 13:37, SGM <notifications@github.com<mailto:notifications@github.com>> wrote:


Did same with my second camera, works perfectly.
My steps:

  1.  I format a SD card with FAT32
  2.  Put the image tf_recovery.img on it.
  3.  Switch off the camera.
  4.  Put the SD card in the camera and power it on.
  5.  Switch off the camera.
  6.  Copy everything from https://github.com/ghoost82/mijia-720p-hack/tree/master/sdcard to SD card root
  7.  Edited mijia-720p-hack.cfg
  8.  Put the SD card in the camera and power it on.
  9.  When the blue light stopped blinking open http://camera_ip/ (don`t try /cgi-bin/status, its working from the "/")

**Comment by zloybatanik** [comment]
@tomsteenbakkers no image in your comment(
https://www.gearbest.com/ip-cameras/pp_701750.html this cameras I used

**Comment by tomsteenbakkers** [comment]
Ok that was the camera I mean. Thanks. Will try it today.

Tom

On 25 Aug 2018, at 13:48, SGM <notifications@github.com<mailto:notifications@github.com>> wrote:


@tomsteenbakkers<https://github.com/tomsteenbakkers> no image in your comment(
https://www.gearbest.com/ip-cameras/pp_701750.html this cameras I used

**Comment by tomsteenbakkers** [comment]
When do you need to switch off the camera after step 4. Is the light yellow or blue?


On 25 Aug 2018, at 13:37, SGM <notifications@github.com<mailto:notifications@github.com>> wrote:


Did same with my second camera, works perfectly.
My steps:

  1.  I format a SD card with FAT32
  2.  Put the image tf_recovery.img on it.
  3.  Switch off the camera.
  4.  Put the SD card in the camera and power it on.
  5.  Switch off the camera.
  6.  Copy everything from https://github.com/ghoost82/mijia-720p-hack/tree/master/sdcard to SD card root
  7.  Edited mijia-720p-hack.cfg
  8.  Put the SD card in the camera and power it on.
  9.  When the blue light stopped blinking open http://camera_ip/ (don`t try /cgi-bin/status, its working from the "/")

**Comment by chris777c** [comment]
I try to do that and don't work for me : 
 1. I format a SD card with FAT32 
 2. Put the image tf_recovery.img **on the root.** 
 3. Switch off the camera. 
 4. Put the SD card in the camera and power it on **and I wait until blue led stopped blinking**
 5. Switch off the camera. 
 **5'. I see SD card and tf_recovery.img is change name to tf_recovery.img.bak and log folder is create white messages file**
6. Copy everything from https://github.com/ghoost82/mijia-720p-hack/tree/master/sdcard to SD card root 
7. Edited mijia-720p-hack.cfg 
8. Put the SD card in the camera and power it on. 

9. When the blue light stopped blinking open http://camera_ip/ **don't work for me neither ssh nor http  access but telnet access work**

**Comment by sdesalve** [comment]
Same thing for me.
Like @chris777

**Comment by tomsteenbakkers** [comment]
Save for me. When using Fing I see that only port 23 (= Telnet) is open. Other ports are not available.

**Comment by tomsteenbakkers** [comment]
I think it has to do that the camera can not boot from the SD. 

How did you format the SD card (tool, size, etc) and how did you copy the content of the directory SDCARD to the sdcard?

**Comment by sdesalve** [comment]
@tomsteenbakkers I've formatted my SD card with Windows 10

![image](https://user-images.githubusercontent.com/9841662/44629334-f3a43100-a94d-11e8-975a-b3fcc356b589.png)

And copied file with Explorer's drag & drop.

Shoud I use an other way?

**Comment by tomsteenbakkers** [comment]
I was wondering why the steps worked for you and not for me. I did the same steps but no hack is applied. So, maybe I’m doing something different. 

I will search for another solution.

**Comment by jymbob** [comment]
Can't check because I've misplaced my SD card, but it probably needs to be
set as bootable.

**Comment by tomsteenbakkers** [comment]
Ok, I will reformat it and set it to bootable. Any suggestions on how to put the content of the directory sdcard on the sdcard. Just copy it using the Mac OS finder?

**Comment by jymbob** [comment]
Yeah, just copying should be fine

**Comment by tomsteenbakkers** [comment]
When I want to create a bootable sdcard and need to select an image file. Which one should I use, the tf_recovery.img from step 1?

**Comment by jymbob** [comment]
No, you're confusing a bootable drive (eg with Linux on) and a Fat32 drive
with the boot flag set. Sorry, no idea how to do that on MacOS

**Comment by tomsteenbakkers** [comment]
Ok, I have format the SD card with the same settings as sdesalve but nothing happens when rebooting the camera. The only difference is that the card is shown in the Mi Home App. I'm using an 4 GB sd card. That the difference between sdesalve. Any suggestions?

**Comment by sdesalve** [comment]
@tomsteenbakkers @jymbob @chris777c  I'm not able to run RTSP server like @zloybatanik stated.

I've followed all steps in this comment https://github.com/samtap/fang-hacks/issues/235#issuecomment-415963279 without any luck.

Is there any other user that have been able to run ~~this hack~~ RTSP server on Chuangmi?

**Comment by tomsteenbakkers** [comment]
@sdesalve sorry I thought it was working for you. I stop trying. Tried everything.

**Comment by jymbob** [comment]
I've got some time in September when I can try it. Will report back

**Comment by tomsteenbakkers** [comment]
Much appreciated!

If I need to test something for you please send me a message and I will help you.

**Comment by jymbob** [comment]
Found some time, opened some wine, and figured this out :smile: 

The issue is that the latest master **doesn't contain the pre-built binaries**

The binaries are on the release track: https://github.com/ghoost82/mijia-720p-hack/releases/tag/v0.95

So, in short:

1. set up camera in MiHome app
2. in camera settings find your IP address
3. power off camera
4. format SD card to FAT32
5. copy tf_recovery.img to the card, insert into camera and power on
6. after about a minute you should have telnet access (`telnet 192.168.1.xxx`)
7. power off camera
7. go to the Ghoost82 repo and download the zip (this is the difference)
8. unpack the zip and copy the contents to the SD card
9. change settings in mijia-720p-hack.cfg
10. insert back into camera and power up
11. after about 30 seconds browse to http://192.168.1.xxx to access your camera

Phew. I'm seeing a working RTSP stream with about a 2 second delay. Thanks all!

I'll tidy this lot up and stick it in my repo when I get chance. Just about to leave for holiday, so probably not before September.
NB: I claim no credit for the code, just doing some caretaking.

**Comment by tomsteenbakkers** [comment]
I will test this new version tomorrow! And will record every step, if this works we have a manual.

**Comment by chris777c** [comment]
thank you very much, great work for me

**Comment by sdesalve** [comment]
It works!

There is some lag and RTSP stream is not very stable... 
I hope in future developements

thank you!

**Comment by tomsteenbakkers** [comment]
It works also for me. I got the RTSP stream! Now trying to add the camera to Homebridge!

**Comment by tomsteenbakkers** [comment]
Just added the camera as homebridge camera, and also this works. I can now use the camera in the Apple Home app. With this hack it's also HomeKit compatible 👍👍

The speed can use some improvement, there is little delay. But, I'm Happy

**Comment by roboterm** [comment]
>Just added the camera as homebridge camera, and also this works. I can now use the camera in the Apple Home app. With this hack it's also HomeKit compatible 👍👍 ...

@tomsteenbakkers can your share the config.json you’re using, pls?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by tomsteenbakkers** [comment]
@roboterm This is the section for the camera

```
        {
            "platform": "Camera-IP",
            "cameras": [
                {
                    "name": "Meterkast",
                    "videoConfig": {
                        "source": "-re -i rtsp://192.168.1.193:554/live/ch00_0",
                        "stillImageSource": "-i http://192.168.1.193/media",
                        "maxStreams": 2,
                        "maxWidth": 640,
                        "maxHeight": 480,
                        "maxFPS": 30
                    }
                }
            ]
        },
```

**Comment by jymbob** [comment]
I've now updated and renamed my repo: https://github.com/jymbob/chuangmi-720-hack

I'm not the author of any of the work, but it seems sensible to collate it in one place rather than on another ticket. Marking this issue closed as we now have a functioning RTSP stream

**Comment by alexandreberton** [comment]
Works also for me great thanks.
The only point is I cannot disable the status light. I tried from the config file and from the web server page but it doesn't work.

**Comment by mtsardakas** [comment]
Hey folks, thanks a lot for this. Has anybody tried it for the 1080p version of the same camera?

**Comment by roboterm** [comment]
@tomsteenbakkers
I have tried your config and I suggest you to add a video codec like omx.

**Comment by cherylldc** [comment]
why my cam its nothing change when step  hack for rstp? 
.img was succes, its show telnet work, but next step seem nothing change, any suggestion?

**Comment by tomsteenbakkers** [comment]
Use this thread https://github.com/miicam/MiiCam and you will also get a user interface op the camera.

**Comment by cherylldc** [comment]
@tomsteenbakkers tq, now my rstp work, but i not to understain about config setting, now my cam cannot view from miapp again, but its work with rtsp, thanks alot, (can i change my config setting on my sd card after this?)

**Comment by tomsteenbakkers** [comment]
@cherylldc When applying the hack the camera doet not work with the Mi Home App. Just remove the SD card, reboot the camera and it's working in the Mi Home App.

Please read this https://miicam.github.io and you will find answers to all your questions.

**Comment by cherylldc** [comment]
ok, thanks alot tom, its very helpfull for me, soo i can use this cam via rtp, and ueonviver/ipcentcom at my windows
